external/RSL_LITE/c_code.c

```diff
@@ -44,7 +44,12 @@ void RSL_LITE_ERROR_DUP1 ( int *me )
 
 /* redirect standard out*/
 # ifndef RSL0_ONLY
-    sprintf(filename,"rsl.out.%04d",*me) ;
+#ifdef LARGE_NUM_CORES 
+    sprintf(filename,"rsl.out.%06d",*me) ;
```

Comment:
From where `rsl_error_dup1` is called in `module_dm.F`, we do have the total number of tasks. We could consider just passing `ntasks` to this routine, then setting the number of digits in the rsl filenames accordingly. We do something similar in MPAS: https://github.com/MPAS-Dev/MPAS-Model/blob/master/src/framework/mpas_log.F#L194-L209 .